src/EXAMPLE_IMPLEMENTATION/java/etithespirit/UnitTesting.java

```diff
@@ -0,0 +1,62 @@
+package etithespirit;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.entity.EntityEmittedSoundEvent;
+import net.minecraftforge.event.RegistryEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.InterModComms;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent;
+import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;
+import net.minecraftforge.fml.event.lifecycle.InterModEnqueueEvent;
+import net.minecraftforge.fml.event.lifecycle.InterModProcessEvent;
+import net.minecraftforge.fml.event.server.FMLServerStartingEvent;
+import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+import net.minecraft.entity.monster.CreeperEntity;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.SoundEvent;
+import net.minecraft.entity.Entity;
+
+
+import java.util.stream.Collectors;
+
+// The value here should match an entry in the META-INF/mods.toml file
+@Mod("unittesting")
+public class UnitTesting
+{
+    private static final SoundEvent SAWTOOTH_WAVE = new SoundEvent(new ResourceLocation("unittesting:implementation.sawtooth"));
+    
+    static {
+        SAWTOOTH_WAVE.setRegistryName("unittesting", "implementation.sawtooth");
+    }
+    
+    public UnitTesting() 
+    {
+        MinecraftForge.EVENT_BUS.register(this);
+        MinecraftForge.EVENT_BUS.register(Implementation.class);
+        GameRegistry.findRegistry(SoundEvent.class).register(SAWTOOTH_WAVE);
```

Comment:
Please remove the use of that `Implementation` class; the event listener should be placed in the outermost class.
Remove the `GameRegistry.findRegistry` call, and replace (as stated above) with using `DeferredRegister` and `RegistryObject`.

All test mods should have a `private static final boolean ENABLE = true;` that controls if the event bus registration calls should be called (using an `if`-check).
This is used for disabling specific tests, such as for executing only a single test or default disabling tests which cause ~~horrible~~ unwanted side-effects in other tests. _(in your case, it can be default enabled, since it does not have side-effects)_